File: source_funcs/go_transcoder/NUMBER_DIGITS_PRODUCT_TWO_NUMBERS/NUMBER_DIGITS_PRODUCT_TWO_NUMBERS.go
package main

import "math"

func f_gold(a int, b int) int {
	var (
		count int = 0
		p     int = int(math.Abs(float64(a * b)))
	)
	if p == 0 {
		return 1
	}
	for p > 0 {
		count++
		p = p / 10
	}
	return count
}

File: rust_funcs/cpp_transcoder/NUMBER_DIGITS_PRODUCT_TWO_NUMBERS/NUMBER_DIGITS_PRODUCT_TWO_NUMBERS.rs
fn min(x: i32, y: i32) -> i32 {if x < y { x } else { y } }
fn max(x: i32, y: i32) -> i32 {if x > y { x } else { y }}
fn cmpfunc(a: &i32, b: &i32) -> std::cmp::Ordering {a.cmp(b)}
fn len(arr: &[i32]) -> usize {arr.len()}
fn sort(arr: &mut [i32]) {arr.sort_by(cmpfunc);}

use std::fs; 
use std::io::prelude::*;
fn f_gold(a: i32, b: i32) -> i32 {     
let mut count = 0;     
let mut p = (a * b).abs();     
if p == 0 {return 1;}    
while p > 0 {        
count += 1;        
p /= 10; 
}
count
}
